$NetBSD: patch-ag,v 1.4 2026/05/19 18:38:10 spz Exp $

deal with ctype functions faulting on invalid input on NetBSD 11

add more 8bit character encodings

--- chset.c.orig	2005-03-30 19:34:27.000000000 +0000
+++ chset.c
@@ -6,7 +6,7 @@
  */
 
 #include <string.h>
-#include <ctype.h>
+#include "ctypeconv.h"
 #include "config.h"
 #include "chset.h"
 
@@ -22,6 +22,8 @@ static struct chset chsets[] = {
     "iso-8859-8", 8,
     "iso-8859-9", 8,
     "iso-8859-15", 8,
+    "koi8-r", 8,
+    "utf-8", 8,
     "unknown", 0,
     NULL, 0,
 };
@@ -35,8 +37,8 @@ getchset(char *name)
     char           *sp;
 
     for (sp = name; *sp; sp++)
-	if (isupper(*sp))
-	    *sp = tolower(*sp);
+	if (ISUPPER(*sp))
+	    *sp = TOLOWER(*sp);
 
     for (csp = chsets; csp->cs_name != NULL; csp++) {
 	if (strcmp(csp->cs_name, name) == 0)
